### PR TITLE
Fix readme

### PR DESCRIPTION
**Pull request by vyorkin**
First, I want to say thank you for this awesome package :+1: 

## What does this pull request do?

I've just tried to follow the "overview" section and noticed a couple of inconsistencies: a couple of copy/paste typos and a few missing types. It may be a bit hard to follow without knowing the exact types of `ValidationError`, `Encrypted` etc, so I've added them. Also, I've added `validateNonEmpty` and `validateMinLength` functions, so now it should be kinda self-contained and easier to follow.

## Other Notes:

Maybe it makes sense to include the code from the Overview in examples? Here is a gist: https://gist.github.com/vyorkin/9dabae99114a5604717f986be36ff31b
Another thing that might be confusing for newcomers like me is the `identity`-trick, I mean `validator = applyOnInputFields (identity { name: V.nonEmpty, ... })`

### CODE REVIEW

**Comment by vyorkin** [comment]
I'll update the PR to include the expandable blocks with full types and validation functions as we've discussed in Slack. Some examples below.

-----

First example:

```purescript
type Errors = Array ValidationError

newtype Form f = Form
  { name      :: f Errors String String
  , password1 :: f Errors String Encrypted
  , password2 :: f Errors String Encrypted
  }
derive instance newtypeForm :: Newtype (Form f) _
```

<details>
  <summary>Expand to see the definition of <code>ValidationError</code> and <code>Encrypted</code> types</summary>

```purescript
newtype Encrypted = Encrypted String

data ValidationError
  = Required
  | TooShort Int
  | EncryptionFailed

derive instance genericVaidationError :: Generic ValidationError _
instance showValidationError :: Show ValidationError where show = genericShow
```

</details>
<br/>
Second example:

```purescript
import Polyform.Validation (Validation(..), hoistFnV)

-- An example Polyform validator
validateEncrypt :: forall m. Monad m => Validation m Errors String Encrypted
validateEncrypt = hoistFnV \str ->
  if null str
     then Invalid [EncryptionFailed]
     else pure (Encrypted str)
```

We can provide a field validation function like this to every field that will be in our form, and then use the `applyOnInputFields` helper function from Formless to convert these functions to run on the `input` field and store their result in the `result` field:

```purescript
import Formless.Validation.Polyform (applyOnInputFields)

-- Each validator in the record should have the type `Validation m error input output`
validator :: forall m. MonadEffect m => Form F.InputField -> m (Form F.InputField)
validator = applyOnInputFields $ identity
  { name: validateNonEmpty
  , password1: validateMinLength 8 *> validateEncrypt
  , password2: validateMinLength 8 *> validateEncrypt
  }
```

<details>
  <summary>Click to see how <code>validateNonEmpty</code> and <code>validateMinLength</code> are defined</summary>

```purescript
validateNonEmpty :: forall m. Monad m => Validation m Errors String String
validateNonEmpty = hoistFnV \str ->
  if null str
     then Invalid [Required]
     else pure str

validateMinLength :: forall m. Monad m => Int -> Validation m Errors String String
validateMinLength n = hoistFnV \p ->
  let p' = String.length p
  in if p' < n
     then Invalid [TooShort n]
     else pure p
```

</details>

**Comment by vyorkin** [comment]
Ready for review (I'll squash the commits at the end)

File: readme.md [hunk 1]
```diff
@@ -179,25 +215,26 @@ Formless manages validation and failed submit attempts on your behalf, only noti
 -- A type representing only the successful parsed values in your Form type
 newtype OutputField error input output = OutputField output
 
--- `unwrapOutputs` is a helper function that will unwrap all these newtypes on your
--- behalf. Used on our custom Form type, it'd apply this transformation:
-unwrapOutputs' :: Form F.OutputField -> { name :: String, password1 :: Encrypted, password2 :: Encrypted }
-unwrapOutputs' = F.unwrapOutputs
+-- `unwrapOutput` is a helper function that will unwrap all these newtypes on your behalf.
+-- Used on our custom Form type, it'd apply this transformation:
+unwrapOutput' :: Form F.OutputField -> FormOutput
```
Comment (thomashoneyman) [inline readme.md]: `FormOutput` isn't actually a type that's been defined in this README. I think I'd like to keep it as the record, no type alias, to make it clear that the output is just the fields with their output values at the end, nothing fancier than that. What do you think?

File: readme.md [hunk 2]
```diff
@@ -158,13 +174,33 @@ import Formless.Validation.Polyform (applyOnInputFields)
...
+validateMinLength :: ∀ m. Monad m => Int -> Validation m Errors String String
+validateMinLength n = hoistFnV \p ->
+  let p' = String.length p
+  in if p' < n
+     then Invalid [TooShort n]
+     else pure p
+```
+
+</details>
+
```
Comment (thomashoneyman) [inline readme.md]: Mind adding an extra newline here? Looks a bit squashed in the markdown viewer.

**Review by thomashoneyman — COMMENTED** [review]
Thanks for cleaning this up, @vyorkin! I have a couple comments in here, but otherwise this should be good to merge. 

I noticed this is under the @vyorkin-forks account. Do you want to open a different PR under your @vyorkin account so that handle is listed as a contributor to the repo instead? Up to you.

**Comment by vyorkin** [comment]
ah, I think its ok, I hope I'll be able to make more valuable contributions in the future :)

**Comment by vyorkin** [comment]
Squashed commits, I think its ready

**Review by thomashoneyman — APPROVED** [review]

**Comment by thomashoneyman** [comment]
Thanks!
